modules/tinymce/src/themes/silver/main/ts/ui/header/InlineHeader.ts

```diff
@@ -126,10 +128,50 @@ export const InlineHeader = (
         Math.max(targetBounds.y - Height.get(container.element) + offset, 0) :
         targetBounds.bottom;
 
-      Css.setAll(mainUi.outerContainer.element, {
+      const baseProperties = {
         position: 'absolute',
-        top: Math.round(top) + 'px',
-        left: Math.round(targetBounds.x) + 'px'
+        left: Math.round(targetBounds.x) + 'px',
+        top: Math.round(top) + 'px'
+      };
+
+      const widthProperties = optToolbarWidth.map(
+        (toolbarWidth: number): {} | { width: string } => {
```

Comment:
This pattern to me looks like where an optional should be instead of `{} | { width: string }` I would suggest the return of this function be `Optional<{width: string}>` and use `bind` function instead of `map`

Reply:
I considered doing about that, but I feel it's a very messy way of doing it. Not that mine feels the best either. Not sure which one people would consider the easiest to read.

Reply:
You're already inside an optional, so you've just created `Optional<{} | { width: string }>`

But when I see code like this
```
          if (availableWidth > width) { // If there's already enough space, don't add a width for performance reasons.
            return {};
          }

          return {
            width: width + 'px'
          };
```

That's an optional. Please change `return {}` to `return Optional.none()` and combine this with the outer optional - the empty case is already handled at the end via `.getOr({ });`

Personally I'd consider keeping it as an `Optional<number>` and using better techniques to apply it than object spread inside `Css.setAll`, but I suspect it's a little too late for that.

Reply:
I am aware that it's in an optional. I just didn't think it'd help readability to add that there. But since there seems to be strong preference for this, I'll do that.

Reply:
`Optional<{} | { width: string }>` is _much_ harder to follow than `toolbarWidth.bind()`, I would hope?